apps/projects/insights.py

```diff
@@ -68,16 +68,41 @@ def create_insight(project: Project) -> ProjectInsight:
     insight, _ = ProjectInsight.objects.get_or_create(project=project)
 
     insight.comments = comments.count()
-    insight.unregistered_participants = unregistered_participants
     insight.ratings = sum(x.count() for x in rating_objects)
     insight.written_ideas = sum(x.count() for x in idea_objects)
     insight.poll_answers = votes.count() + answers.count()
     insight.live_questions = live_questions.count()
     insight.save()
 
     insight.active_participants.clear()
+    unregistered_participants = set()
+
     for obj in creator_objects:
-        ids = list(obj.values_list("creator", flat=True).distinct().order_by())
+        # ignore objects which don't have a creator, they are counted in the next step.
+        ids = list(
+            obj.filter(creator__isnull=False)
+            .values_list("creator", flat=True)
+            .distinct()
+            .order_by()
+        )
         insight.active_participants.add(*ids)
+        # content from unregistered users doesn't have a creator but a content_id
+        if model_field_exists(obj.model, "content_id"):
+            content_ids = set(
```

Comment:
I now added more generic way of counting the unregistered participants, what do you think ? If that's fine I'd squash this commit and merge it afterwards